dags/sentinel-2_gap_filler.py

```diff
@@ -0,0 +1,182 @@
+            "DataType": "String",
+            "StringValue": datetime,
+        }
+
+    cloudcover = stac_doc.get("properties").get("eo:cloud_cover")
+    if cloudcover:
+        msg_attributes["cloudcover"] = {
+            "DataType": "Number",
+            "StringValue": str(cloudcover),
+        }
+
+    maturity = stac_doc.get("properties").get("dea:dataset_maturity")
+    if maturity:
+        msg_attributes["maturity"] = {
+            "DataType": "String",
+            "StringValue": maturity,
+        }
+
+    bbox = stac_doc.get("bbox")
+    if bbox and len(bbox) > 3:
+        msg_attributes["bbox.ll_lon"] = {
+            "DataType": "Number",
+            "StringValue": str(bbox[0]),
+        }
+        msg_attributes["bbox.ll_lat"] = {
+            "DataType": "Number",
+            "StringValue": str(bbox[1]),
+        }
+        msg_attributes["bbox.ur_lon"] = {
+            "DataType": "Number",
+            "StringValue": str(bbox[2]),
+        }
+        msg_attributes["bbox.ur_lat"] = {
+            "DataType": "Number",
+            "StringValue": str(bbox[3]),
+        }
+
+    return msg_attributes
+
+
+def get_missing_stac_files(offset=0, limit=None):
+    """
+    read the gap report
+    """
+
+    hook = S3Hook(aws_conn_id=dag.default_args["africa_conn_id"])
+    bucket_name, key = hook.parse_s3_url(default_args["s3_report_path"])
+    print(f"Reading  {default_args['s3_report_path']}")
+
+    s = time.time()
+    files = (
+        hook.read_key(key=key, bucket_name=bucket_name)
+        .replace("sentinel-cogs-inventory", f"{default_args['src_bucket_name']}")
+        .splitlines()
+    )
+
+    print(f"Read took {time.time()-s} seconds")
+    for f in files[offset:limit]:
+        yield f
+
+
+def publish_messages(messages):
+    """
+    Publish messages to the data transfer queue
+    param message: list of messages
+    """
+
+    sqs_hook = SQSHook(aws_conn_id=dag.default_args["us_conn_id"])
+    sqs = sqs_hook.get_resource_type("sqs")
+    queue = sqs.get_queue_by_name(QueueName=default_args.get("queue_name"))
+    queue.send_messages(Entries=messages)
+
+    return []
+
+
+def get_contents_and_attributes(hook, s3_filepath):
+    bucket_name, key = hook.parse_s3_url(s3_filepath)
+    contents = hook.read_key(key=key, bucket_name=default_args["src_bucket_name"])
+    contents_dict = json.loads(contents)
+    attributes = get_common_message_attributes(contents_dict)
+    return contents, attributes
+
+
+def prepare_and_send_messages():
+
+    hook = S3Hook(aws_conn_id=dag.default_args["us_conn_id"])
+    # Read the missing stac files from the gap report file
+    files = get_missing_stac_files(0, 2)
+
+    counter = 0
+    messages = []
+    # counter for files that no longet exist
+    failed = 0
+    for s3_path in files:
+        key_exists = hook.check_for_key(s3_path)
+        if not key_exists:
+            failed += 1
+            raise ValueError(f"{s3_path} does not exist")
```

Comment:
This will end the processing of the whole thing, not just this one file

Reply:
Have moved all of this to a function and added a statement to handle the exception here.